syntax = "proto3";
package chasky;

enum DataType {
  // Not a legal type, just a placeholder
  CH_INVALID = 0;

  // All basic types supported
  CH_FLOAT = 1;
  CH_INT64 = 2;
  CH_UINT64 = 3;
  CH_STRING = 4;

  // a placeholder for functions that works for any datatype.
  CH_COMMON = 8;

  // All matrixs
  CH_MAT_FLOAT = 11;

  // Variadic number of arguments
  CH_MAT_FLOAT_LIST = 21;

  // All matrix types supported
  // ...
}